packages/ra-ui-materialui/src/input/TimeInput.tsx

```diff
@@ -0,0 +1,154 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { useInput, FieldTitle } from 'ra-core';
+
+import { CommonInputProps } from './CommonInputProps';
+import { sanitizeInputRestProps } from './sanitizeInputRestProps';
+import { InputHelperText } from './InputHelperText';
+
+/**
+ * Converts a time string without timezone to a date object
+ * with timezone, using the browser timezone.
+ *
+ * @param {string} value Date string, formatted as hh:mm
+ * @return {Date}
+ */
+const parseTime = (value: string) => {
+    const today = new Date();
+    return value ? new Date(today.toLocaleDateString() + ' ' + value) : value;
+};
```

Comment:
```suggestion
const parseTime = (value: string) => {
    if (!value) return value;
    const timeTokens = value.split(':').map(v => parseInt(v));
    const today = new Date();
    today.setHours(timeTokens[0] ?? 0);
    today.setMinutes(timeTokens[1] ?? 0);
    return today;
};
```

Reply:
Here it is! 🙈

Reply:
What was the source of the issue? Was the `toLocaleDateString` function not supported on your machine? Was the string given to the `Date` object not understood?

Seems strange it would fail only on some macos devices.

Anyways your fix looks good and achieves the same functionality without complicating the code too much so let's use this!

I can't make the commit myself right now so feel free to push to the PR.

Thanks for fixing this!

Reply:
The error was nesting in the Date parsing which was returning an invalid date.
`new Date(today.toLocaleDateString() + ' ' + value)` => `new Date('18/08/2022 10:22')`

Reply:
Unfortunately it appears we do not have push permission on your branch @arnaudvergnet :slightly_frowning_face: 

Could you commit these changes when you have time?

Again, sorry for taking so long, and thank you so much for your patience. :slightly_smiling_face:

Reply:
> it was nested in the Date parsing which was returning an invalid date.

Oh maybe the parsing failed because you had your computer locale in  French so the date parsing behaved differently from mine (my computer was in English).

Reply:
> Could you commit these changes when you have time?

Can't find the option to allow edits by maintainers and I cannot make the commit right now so this will have to wait a few days if that's okay with you.

Reply:
> > Could you commit these changes when you have time?
> 
> Can't find the option to allow edits by maintainers and I cannot make the commit right now so this will have to wait a few days if that's okay with you.

That's perfectly fine. :slightly_smiling_face: 
Besides, how could we urge you when we ourselves took ages to review your PR in the first place :joy: